OData/src/System.Web.OData/OData/Query/Expressions/SelectExpandBinder.cs

```diff
@@ -391,7 +391,7 @@ private Expression CreateTotalCountExpression(Expression source, ExpandedNavigat
                 Expression propertyName = CreatePropertyNameExpression(elementType, propertyToExpand, source);
                 Expression propertyValue = CreatePropertyValueExpressionWithFilter(elementType, propertyToExpand, source,
                     expandItem.FilterOption);
-                Expression nullCheck = Expression.Equal(propertyValue, Expression.Constant(null));
+                Expression nullCheck = GetNullCheckExpression(propertyToExpand, propertyValue, projection);
```

Comment:
@postoronnimw Yes, I mean you have changed the original behavior to support NHibernate, can  we change the behavior depends on the LINQ provider, for example, is the provider is EF, keep the old behavior, if it is NHibernate, use the new expression